src/procstat.c

```diff
@@ -731,25 +732,46 @@ int procstat_create_simple(struct procstat_context *context,
 	return -1;
 }
 
-void procstat_u64_series_add_point(struct procstat_series_u64 *series, uint64_t value)
+bool is_reset(struct procstat_series_u64 *series)
 {
-	int64_t delta;
-	int64_t delta2;
-	int64_t avg_delta;
-	unsigned reset;
-
+	uint64_t reset_interval, time_since_last_reset;
+	unsigned reset = 0;
+	bool reset_interval_expired = false;
+
+	struct timespec cur_time;
+	if (clock_gettime(CLOCK_REALTIME, &cur_time) == 0) {
+		time_since_last_reset = cur_time.tv_sec - series->last_reset_time;
+		reset_interval = __atomic_load_n(&series->reset_interval, __ATOMIC_RELAXED);
+		if ((reset_interval) && (time_since_last_reset > reset_interval)) {
+			reset_interval_expired = true;
+			series->last_reset_time = cur_time.tv_sec;
+		}
+	}
 	/* We dont care about ordering of reset with another thread
-	 * we dont care if because of some raise we will do 2 resets */
-	reset = __atomic_load_n(&series->reset, __ATOMIC_RELAXED);
-	if (reset) {
+	* we dont care if because of some race we will do 2 resets */
+	if (reset_interval_expired == false) reset = __atomic_load_n(&series->reset, __ATOMIC_RELAXED);
```

Comment:
why one liners for the if else command? This is not the recommended coding style

Reply:
done